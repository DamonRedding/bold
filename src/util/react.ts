import { MutableRefObject, Ref } from 'react'

export function composeRefs<T>(...refs: Array<Ref<T>>) {
    return (value: T) => refs.forEach(ref => setRef(ref, value))
}

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}

/**
 * Returns the next sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getNextSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.nextElementSibling) {
        current = current.nextElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}

/**
 * Returns the previous sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getPreviousSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.previousElementSibling) {
        current = current.previousElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}
