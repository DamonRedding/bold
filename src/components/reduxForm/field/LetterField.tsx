import * as React from 'react'
import { TextField } from './TextField/TextField'

export const normalize = value => value.replace(/[^A-Za-záéíóúÁÉÍÓÚçÇâêôÂÊÔõãÕÃäöüÄÖÜàÀ' ]/g, '')

export interface LetterFieldProps {
    name: string
}

export class LetterField extends React.Component<LetterFieldProps> {

    render() {
        return (
            <TextField {...this.props} normalize={normalize} />
        )
    }

}
