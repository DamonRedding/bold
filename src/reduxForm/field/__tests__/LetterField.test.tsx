import * as React from 'react'
import { mount } from 'enzyme'
import { normalize, LetterField } from '../LetterField'
import { shallowRenderAndMatch, testForm } from '../../../__tests__/test.defaults'

describe('LetterField', () => {
    describe('normalize', () => {
        it('deve aceitar somente letras e espaços', () => {
            expect(normalize('21 abc 42')).toEqual(' abc ')
        })
    })

    describe('LetterField', () => {
        it('render', () => {
            shallowRenderAndMatch(testForm(() => <LetterField name='letter' />))
        })

        it('deve aceitar somente letras e espaços', () => {
            const wrapper = mount(testForm(() => <LetterField name='letter' />))
            const inputWrapper = wrapper.find('[name="letter"]')
            inputWrapper.simulate('change', { target: { value: 'abc 123a!' } })
            expect(inputWrapper.props().value).toEqual('abc a')
        })
    })
})
