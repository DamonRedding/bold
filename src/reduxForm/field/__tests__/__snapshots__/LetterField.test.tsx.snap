// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterField LetterField render 1`] = `<ReduxForm />`;
